api/controllers/v0.3/endpoints.js

```diff
@@ -351,7 +352,106 @@ async function createPatron(req, res) {
   renderResponse(req, res, response.status, response);
 }
 
+/**
+ * setupDependentEligibility(req, res)
+ * The callback for the "/api/v0.3/validations/username" route. This will make
+ * sure that everything is set up correctly in order to make a request to
+ * the ILS. This includes validating environment variables and decrypting
+ * AWS keys.
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ */
+async function setupDependentEligibility(req, res) {
+  const hasValidEnvVariables = await checkEnvVariables(
+    req,
+    res,
+    ROUTE_TAG,
+    envVariableNames
+  );
+
+  if (!hasValidEnvVariables) {
+    // `checkEnvVariables` already sent the error response so return the
+    // function so it doesn't continue to process the current request.
+    return;
+  }
+
+  ilsClientKey = process.env.ILS_CLIENT_KEY;
+  // ilsClientKey || awsDecrypt.decryptKMS(process.env.ILS_CLIENT_KEY);
+  ilsClientPassword = process.env.ILS_CLIENT_SECRET;
+  // ilsClientPassword || awsDecrypt.decryptKMS(process.env.ILS_CLIENT_SECRET);
+
+  Promise.all([ilsClientKey, ilsClientPassword])
+    .then((decryptedValues) => {
+      [ilsClientKey, ilsClientPassword] = decryptedValues;
+      checkDependentEligibility(req, res);
+    })
+    .catch(() => {
+      const localErrorMessage =
+        "The ILS ClientKey and/or Secret were not decrypted";
+
+      const errorResponseData = modelResponse.errorResponseData(
+        collectErrorResponseData(
+          500,
+          "configuration-error",
+          localErrorMessage,
+          "",
+          ""
+        ) // eslint-disable-line comma-dangle
+      );
+      renderResponse(req, res, 500, errorResponseData);
+    });
+}
+
+/**
+ * checkDependentEligibility(req, res)
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ */
+async function checkDependentEligibility(req, res) {
+  const timeNow = new Date();
+  // eslint-disable-next-line max-len
+  const ilsTokenExpired =
+    ilsTokenTimestamp && timeNow - ilsTokenTimestamp > 3540000; // 3540000 = 59 minutes; tokens are for 60 minutes
+  if (!ilsToken || ilsTokenExpired) {
+    getIlsToken(req, res, ilsClientKey, ilsClientPassword).then(() => {
+      checkDependentEligibility(req, res);
+    });
+    return;
+  }
+
+  ilsClient =
+    ilsClient ||
+    IlsClient({
+      createUrl: process.env.ILS_CREATE_PATRON_URL,
+      findUrl: process.env.ILS_FIND_VALUE_URL,
+      ilsToken,
+      tokenUrl: process.env.ILS_CREATE_TOKEN_URL,
+      // ilsTokenTimestamp,
```

Comment:
Were these supposed to be commented out...?

Reply:
Yeah, I haven't refactored this part yet. I want to try passing in these values or setting them in the `IlsClient` itself, but since those are variables that the whole app cares about, it seems to be better to leave them at the app scope rather than in the class scope. It's also related to how the lambda startups and runs.